### PR TITLE
Integrate dev.to blogs into the website

### PR DESCRIPTION
**Pull request by cherylli**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**152**](https://github.com/MarianaSouza/web-dev-path/issues/152) |

closing #152 

#### Have you updated the CHANGELOG.md file? If not, please do it.
yes

#### What is this change?
- Updated the blog page according to the design. Content is pulled from dev.to webdevpath organization.
- Updated Card components to work with blog post display
- Added routes for blog pages (all blogs, tags dynamic routes) (view all)

#### Were there any complications while making this change?
1. Sometimes blog post titles are too long, temporary solution - truncate to 2 lines as per design
   This is probably not desired, some possible alternatives are
   a. make the title font smaller
   b. only show the title and tags
   c. make the card longer?

2. I forgot view all funcationality is not require but I've already implemented it but since we don't have a design it's mostly just displaying the cards and title

3. There's only 3 posts at the moment, so I copied the array a few times in the "view all" page for all posts. Also not sure if you want them to be included in the live website after this is approved

4. There's no design for rows with less than 3 cards, so I implemented it this way
image

#### How did you verify this change?

run locally

#### When should this be merged?
After review

### CODE REVIEW

**Comment by mariana-caldas** [comment]
Hey @cherylli ! This PR is looking great!
I've got some points[ here](https://www.loom.com/share/a3fdbd2d5ee2493ca7c055a7061733e0) I think we need to refine. Please let m know if that makes sense 🚀

**Comment by cherylli** [comment]
Hi, search is not working yet as intended as per issue description.

The view all without swipe is also intended as I think it might look weird if there are more posts. Since they are all in the same category, if we group them in 3s. 

I mean they are the same thing so either swipe them all or not swipe at all, but if we swipe them all, the page will be very short.

Hope that makes sense. But I'm happy to change it if that's what you want. 

The width, I tried but didn't seem to work, I forgot what the issue was. I think it has something to do with the swiper. I think with 2 cards it always 'space between' regardless of width. I might try again with a smaller width, and see if I can change the justify setting

**Comment by mariana-caldas** [comment]
> Hi, search is not working yet as intended as per issue description.
> 
> The view all without swipe is also intended as I think it might look weird if there are more posts. Since they are all in the same category, if we group them in 3s.
> 
> I mean they are the same thing so either swipe them all or not swipe at all, but if we swipe them all, the page will be very short.
> 
> Hope that makes sense. But I'm happy to change it if that's what you want.
> 
> The width, I tried but didn't seem to work, I forgot what the issue was. I think it has something to do with the swiper. I think with 2 cards it always 'space between' regardless of width. I might try again with a smaller width, and see if I can change the justify setting

Hey @cherylli !

* Search functionality -> it's not a good practice to send anything unfinished to production. So, as I mention in the video, we can either comment it out for now and finish that in the next PR or try concluding the implementation. I'll let you decide it :)

* Swipe -> The problem I detail in the video, Cheryl, is that it feels weird the swipe functionality disappears after viewing all on mobile devices. I think that, on mobile devices, when you click "View all" to see all the posts from a topic, the swipe functionality should be kept. What do you think?

Please let me know if that makes sense, Cheryl. Thanks!

**Comment by cherylli** [comment]
Let me do both and make some screenshots and post on slack so others can see and comment. I wrote the code in a way I can switch between 2 modes easily

**Comment by cherylli** [comment]
> 

good point. i'll hide the search for now. 

Also are you happy to keep my posts up or do you want me to remove them? I'm not sure what type of blog posts should go here (on live)

**Comment by mariana-caldas** [comment]
> > 
> 
> good point. i'll hide the search for now.
> 
> Also are you happy to keep my posts up or do you want me to remove them? I'm not sure what type of blog posts should go here (on live)

Thanks, @cherylli !
I talk about some of the points you mention on Slack in this video too
https://www.loom.com/share/0a3ff7407a5841aabf57f0b5e39795a3

Can you just remove the repeated blog posts, please? Your blog posts are perfect for the project purpose, Cheryl! I'm writing something as well in the next weeks too 🙌🏼

**Comment by cherylli** [comment]
Thanks. In response to your questions.
- I prefer putting formal discussions here and less formal ones on slack
- I reverted the view all page to no horizontal scroll, but I think maybe we should implement something like pagination or infinite scroll when we have more posts in future?
- moved "view all" to the middle in the mobile view
- I fixed the "3 sliders per view" to just show in the blog page (using the same component), which was an oversight on my part

Also fixed a bit of styling issues. Please let me know if there's any issues

File: components/blog/BlogPostsContainer.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            <Container>
+              <div className={styles.postContainer}>
+                {posts.map((p, index) => (
+                  <Card customClass='blog' key={index} card={p} />
+                ))}
+              </div>
+            </Container>
+          )
+        }
+        {viewall ? (
```
Comment (mariana-caldas) [inline components/blog/BlogPostsContainer.js:57]: Can we add "view all" only when we have more than 3 posts, @cherylli , pls?

File: components/containers/CardsColumns.js [hunk 2]
```diff
@@ -10,33 +10,47 @@ import styles from '@/styles/CardsColumns.module.scss';
...
+          pagination
+          centerInsufficientSlides={false}
+          slidesPerView={1}
+          breakpoints={{
+            768: {
+              slidesPerView: 2,
+              spaceBetween: 20,
+            },
+            1334:
+              customClass === 'blog'
```
Comment (mariana-caldas) [inline components/containers/CardsColumns.js:34]: That was a cool workaround, @cherylli ! 🚀

File: components/snippets/Title.js [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+    <h2
+      className={
+        customClass ? `${styles.title} ${styles[customClass]}` : styles.title
+      }
+    >
+      {title}
+    </h2>
+  );
+
+  return customClass === 'blogTitle' ? (
```
Comment (mariana-caldas) [inline components/snippets/Title.js:15]: Can we use the <Container /> directly on the <BlogPostsContainer /> , @cherylli ?
I'd rather keep this Title snippet very simple and with fewer dependencies if possible. Adding a customClass here was a great idea!

**Review by mariana-caldas — APPROVED** [review]
Other than that change on View All, I think we're done, @cherylli !
Amazing job on this feature! 🚀

**Pull request merged**